src/main/kotlin/com/autonomousapps/tasks/AdviceTransformTask.kt

```diff
@@ -0,0 +1,98 @@
+@file:Suppress("UnstableApiUsage")
+
+package com.autonomousapps.tasks
+
+import com.autonomousapps.TASK_GROUP_DEP
+import com.autonomousapps.internal.advice.*
+import com.autonomousapps.internal.utils.*
+import com.autonomousapps.internal.utils.chatter
+import com.autonomousapps.tasks.AdviceTask.ConsoleReport
+import org.gradle.api.DefaultTask
+import org.gradle.api.file.RegularFileProperty
+import org.gradle.api.provider.Property
+import org.gradle.api.tasks.*
+import java.lang.StringBuilder
+
+/**
+ * Produces human- and machine-readable advice on how to modify a project's dependencies in order to have a healthy
+ * build.
+ */
+@CacheableTask
+abstract class AdviceTransformTask : DefaultTask() {
+
+  init {
+    group = TASK_GROUP_DEP
+    description = "Displays advice on screen"
+  }
+
+  @get:PathSensitive(PathSensitivity.RELATIVE)
+  @get:InputFile
+  abstract val adviceConsoleReport: RegularFileProperty
+
+  @get:Input
+  abstract val chatty: Property<Boolean>
+
+  @get:OutputFile
+  abstract val adviceConsoleReportTxt: RegularFileProperty
+
+  private val chatter by lazy { chatter(chatty.get()) }
+
+  @TaskAction
+  fun action() {
+    // Output
+    val adviceConsoleReportTxtFile = adviceConsoleReportTxt.getAndDelete()
+
+    // Inputs
+    val adviceConsoleReportFile = adviceConsoleReport.asFile.get()
+
+    val consoleReport = adviceConsoleReportFile
```

Comment:
I added a companion utility to `getAndDelete()`, in the same file, that looks like:
```kotlin
internal inline fun <reified T> RegularFileProperty.fromJsonSet(): Set<T> {
  return get().asFile.readText().fromJsonSet()
}
```
you could write companions to that that aren't `Set`s.